kernel/packages/shared/session/sagas.ts

```diff
@@ -168,7 +169,7 @@ function* startSignUp(userId: string, identity: ExplorerIdentity) {
   let prevGuest = fetchProfileLocally(userId)
   let profile: Profile = prevGuest ? prevGuest : yield generateRandomUserProfile(userId)
   profile.userId = identity.address.toString()
-  profile.ethAddress = identity.address.toString()
+  profile.ethAddress = identity.rawAddress ? identity.rawAddress : profile.userId
```

Comment:
we either have or not have an eth address, please do not mix pears and bananas under names that may have impact on smart contracts.

What is the difference between userId and rawAddress?

Reply:
The userId always contained the **_lowercased ethereum address_** (stored in `identity.address`), now we store the real ethereum address in `ethAddress`.

Thanks for your suggestion, I removed the whole `rawAddress` existence check as it doesn't make sense (since we first get the raw address and then lowercase it for the userId), and tested everything again.